src/twisted/internet/test/test_iocp.py

```diff
@@ -210,3 +226,44 @@ def cb(self, rc, bytes, evt):
         self.assertEqual(fd.counter, EVENTS_PER_LOOP)
         ir.doIteration(0)
         self.assertEqual(fd.counter, EVENTS_PER_LOOP + 1)
+
+    def test_doubleHalfClose(self):
+        """
+        If one side half-closes its connection, and then the other side of the
+        connection calls C{loseWriteConnection}, and then C{loseConnection} in
+        {writeConnectionLost}, the connection is closed correctly.
+
+        This covers a code branch for ticket U{https://twistedmatrix.com/trac/ticket/9553}.
+        """
+
+        @implementer(IHalfCloseableProtocol)
+        class ListenerProtocol(ConnectableProtocol):
+            def readConnectionLost(self):
+                # Force the existence of a writer in an attempt to cover #9553.
+                self.transport.startWriting()
+                self.transport.loseWriteConnection()
+
+            def writeConnectionLost(self):
+                self.transport.loseConnection()
+
+        class Client(ConnectableProtocol):
+            def connectionMade(self):
+                self.transport.loseConnection()
+
+        # If test fails, reactor won't stop and we'll hit timeout:
+        server = ListenerProtocol()
+        client = Client()
+
+        initial_handles = []
+
+        def reactorSetUp(reactor):
+            reactor._handleSignals()
+            # We might have the waker readers.
+            initial_handles.extend(reactor.handles.copy())
+
+        reactor = runProtocolsWithReactor(
+            self, server, client, TCPCreator(), reactorSetUp
+        )
+
+        # Check that the reactor is clean.
+        self.assertCountEqual(initial_handles, reactor.handles)
```

Comment:
![image](https://user-images.githubusercontent.com/413772/110222398-c430f080-7ec9-11eb-874f-ec7f7ff8245e.png)

still not covered?

Reply:
Looks like windows coverage is not reported.

I have to inspect Codecov, but on coveralls I only see the posix builds... 

![image](https://user-images.githubusercontent.com/204609/110382027-3fbda980-8052-11eb-8ef9-fa93c3c43bca.png)

iocp is zero

![image](https://user-images.githubusercontent.com/204609/110382204-798eb000-8052-11eb-8206-4802d157476e.png)

-------

I will have to check if coveralls has some merge feature... or we can normalize the paths before pushing the reports.

I know that codecov can do that for you

Reply:
same on codecov .... but paths are normalized
![image](https://user-images.githubusercontent.com/204609/110382551-fb7ed900-8052-11eb-87f1-fe462d17edcb.png)